### PR TITLE
ARROW-10271: [Rust] Update dependencies

### PR DESCRIPTION
**Pull request by nevi-me**
Tested by running tests, benchmarks and examples.
Did this with `simd` on and off

The main one here is `packed_simd` > `packed_simd_2`,
as the former is broken on recent nightlies

### CODE REVIEW

**Comment by nevi-me** [comment]
I've marked this as a blocker for 2.0. I also tested with `nightly-2020-10-10`, and tests ran fine.

@andygrove, out of curiousity, is there any reason why datafusion doesn't enable simd by default? Looks like there's no easy way of enabling it, so users could be losing out on the benefits on compute.

**Review by jorgecarleitao — APPROVED** [review]
Thanks a lot for taking this, @nevi-me . I agree that we should fix it ahead of 2.0.0; I have reviewed it and LGTM (pending CI passing)

**Review by andygrove — APPROVED** [review]

**Comment by andygrove** [comment]
@nevi-me @jorgecarleitao I think we're too late to get this into 2.0.0 since the release candidate has already been cut and the voting process has started. I just ran the tests on the release candidate and it looks fine. If I'm understanding correctly, the issue resolved here is only relevant if we move away from the current nightly Rust?

**Comment by nevi-me** [comment]
Thanks Andy, it'll affect anyone who uses 2.0 with SIMD and a recent nightly. I don't yet know when pckdsmd broke, so it could be any nightly in the last month maybe.

**Comment by andygrove** [comment]
Got it. If there is a 2.0.1 release then we could consider applying this patch there.

**Pull request closed without merging**
